### ISSUE TITLE
Introduction of --skip-needs results in breaking change

### ISSUE THREAD

**Issue by strainovic** [helmfile/helmfile#339]
### Operating system

Windows x64

### Helmfile Version

0.143.0-0.145.3

### Helm Version

3.9.0

### Bug description

We use a selector(s) to match and template/sync only a single or subset of services. Everything worked fine until we updated the helmfile version.

`helmfile version v0.142.0` works fine with `--selector/-l` and `template` command

`helmfile version v0.143.0` with the same command produces the following error(s):

```
in ./helmfile.yaml: release(s) "platform-dev/bpm" depend(s) on an undefined release "platform-dev/authentication". Perhaps you made a typo in "needs" or forgot defining a release named "authentication" with appropriate "namespace" and "kubeContext"?
```

The `sync` command worked fine with `-selector/-l` with both mentioned versions.

`Helmfile version 0.145.2`  works fine with `--selector/-l` and `sync` command

`Helmfile version 0.145.3` with the same command produces the following error(s):

```
Error: in ./helmfile.yaml: release "platform-dev/comments" depends on "platform-dev/api-management" which does not match the selectors. Please add a selector like "--selector name=api-management", or indicate 
whether to skip (--skip-needs) or include (--include-needs) these dependencies

in ./helmfile.yaml: release "platform-dev/comments" depends on "platform-dev/api-management" which does not match the selectors. Please add a selector like "--selector name=api-management", or indicate whether to skip (--skip-needs) or include (--include-needs) these dependencies
```

Everything works fine if we add `-skip-needs` on the latest helmfile released version.

Do we really need this breaking change and if yes, what is the benefit of it?

Thanks :)

### Example helmfile.yaml

-

### Error message you've seen (if any)

-

### Steps to reproduce

-

### Working Helmfile Version

0.142.0

### Relevant discussion

-

**Comment by yxxhero** [comment]
@strainovic I will work on this. Thanks for your issue.

**Comment by mumoshu** [comment]
We added this to fix the bug that `-l foo=bar` resulted in unfilled need to be silently ignored. https://github.com/roboll/helmfile/pull/1772
Other folks complained so I changed it to use `--skip-needs` by default in https://github.com/roboll/helmfile/releases/tag/v0.139.2.
Now, I don't understand why it's still happening!

@yxxhero How are you going to "fix" it? Did we unintentionally introduced a regression when we migrated to cobra?

**Comment by mumoshu** [comment]
@yxxhero Which commit/pull request broke it then...? The only relevant commit in v0.143.0 would be https://github.com/roboll/helmfile/pull/2026 and it has been added for a good reason.
@strainovic Hey! Are you sure you don't actually have any typo in your helmfile's `needs` entries? Could you share your helmfile.yaml?

**Comment by mumoshu** [comment]
@yxxhero I reviewed #342 but I'm still unsure what it's supposed to do. Are you going to make `--skip-needs` the default? But I thought we already made it so in https://github.com/roboll/helmfile/releases/tag/v0.139.2

**Comment by yxxhero** [comment]
https://github.com/roboll/helmfile/pull/1835
I see this PR. 
I don't understand why --skip-needs is default true?

from the code. it will return false when --skip-needs not provided.

**Comment by mumoshu** [comment]
@yxxhero I think the key difference is the use of `BoolTFlag` vs `BoolFlag`.

**Comment by yxxhero** [comment]
@mumoshu oh. I konw. I will check the code again.

**Comment by yxxhero** [comment]
@mumoshu done. sorry for this issue.

**Comment by mumoshu** [comment]
@yxxhero Thanks! To be extra clear, your fix addresses the regression introduced in 0.145.0, right?

**Comment by yxxhero** [comment]
@mumoshu yes.
